esmvaltool/cmorizers/data/cmorizer.py

```diff
@@ -0,0 +1,534 @@
+        """
+        if not self.datasets:
+            logger.error('Missing datasets to download')
+        logger.info("Downloading original data...")
+        # master directory
+        failed_datasets = []
+        for dataset in self.datasets:
+            try:
+                self.download_dataset(dataset, start_date, end_date, overwrite)
+            except Exception:
+                logger.exception('Failed to download %s', dataset)
+                failed_datasets.append(dataset)
+        if failed_datasets:
+            logger.error('Download failed for datasets %s', self.datasets)
+            return False
+        return True
+
+    def download_dataset(self, dataset, start_date, end_date, overwrite):
+        """Download a single dataset.
+
+        Parameters
+        ----------
+        dataset: str
+            Dataset name
+        start_date: datetime
+            First date to download
+        end_date: datetime
+            Last date to download
+        overwrite: boolean
+            If True, download again existing files
+        """
+        if not self.has_downloader(dataset):
+            raise ValueError(
+                f'Dataset {dataset} does not have an automatic downloader')
+        dataset_module = self._dataset_to_module(dataset)
+        logger.info('Downloading %s', dataset)
+        logger.debug("Download module: %s", dataset_module)
+        try:
+            downloader = importlib.import_module(
+                f'.{dataset_module}',
+                package='esmvaltool.cmorizers.data.downloaders.datasets')
+        except ImportError:
+            logger.exception('Could not find cmorizer for %s', dataset)
+            raise
+
+        downloader.download_dataset(self.config, dataset,
+                                    self.datasets_info['datasets'][dataset],
+                                    start_date, end_date, overwrite)
+        logger.info('%s downloaded', dataset)
+
+    def format(self, start, end, install):
+        """Format all available datasets.
+
+        Parameters
+        ----------
+        start: datetime
+            Start of the period to format
+        end: datetime
+            End of the period to format
+        install: bool
+            If True, automatically moves the data to the final location if
+            there is no
+        """
+        logger.info("Running the CMORization scripts.")
+        # datasets dictionary of Tier keys
+        datasets = self._assemble_datasets()
+        if not datasets:
+            logger.warning("Check input: could not find required %s in %s",
+                           self.datasets, self.rawobs)
+        logger.info("Processing datasets %s", datasets)
+
+        # loop through tier/datasets to be cmorized
+        failed_datasets = []
+        for dataset in datasets:
+            if not self.format_dataset(dataset, start, end, install):
+                failed_datasets.append(dataset)
```

Comment:
Would it be possible to add somewhere a meesage that a given formatting is finished if that makes sense? For example: `logger.info('%s formatter', dataset)` or `logger.info('%s formatting finished', dataset)`

Reply:
Yes, it is added now